Ukoly/DU_9/class.py

```diff
@@ -0,0 +1,64 @@
+class band_member:
+    def __init__(self, name):
+        self.name = name
+
+    def play(self):
+        print(f'{self.name} plays on {self.instrument}')
+
+
+class guitarist(band_member):
+    def play(self, number_of_strings):
+        self.number_of_strings = number_of_strings
+        if self.number_of_strings >= 4 and self.number_of_strings < 6:
+            self.instrument = 'bass guitar'
+        elif self.number_of_strings >= 6:
+            self.instrument = 'solo guitar'
+        super().play()
+
+
+class singer(band_member):
+    def play(self):
+        print(f'{self.name} sings')
+
+
+class drummer(band_member):
+    instrument = 'drums'
+
+    def __init__(self, name):
+        self.number_of_sticks = 2
+        super().__init__(name)
+
+    def throw_a_stick(self):
+        self.number_of_sticks += 1
+
+    def number_of_throwen_sticks(self):
```

Comment:
Táto celá metóda mi nejak nedáva zmysel, okrem toho posledného výpisu na konci. V prvom kroku máš `if`, ktorý kontroluje, či metóda `.throw_a_stick()` vráti niečo, čo by sa dokázalo vyhodnotiť ako `True`. Avšak metóda `throw_a_stick` je technicky procedúra a jej návratová hodnota je vždy `None`. Čo je vždy `False`, plus ako bonus sa nám pridá jedna vyhodená palička. A nakoniec ešte pridáš ďalšie 2? To nechápem už vôbec.